src/steamship/experimental/package_starters/telegram_bot.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional, Type
+
+import requests
+from pydantic import Field
+
+from steamship import SteamshipError
+from steamship.invocable import Config, InvocableResponse, PackageService, post
+
+
+class TelegramBotConfig(Config):
+    bot_token: str = Field(description="The secret token for your Telegram bot")
+
+
+class TelegramBot(PackageService, ABC):
+
+    config: TelegramBotConfig
+
+    @classmethod
+    def config_cls(cls) -> Type[Config]:
+        """Return the Configuration class."""
+        return TelegramBotConfig
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.api_root = f"https://api.telegram.org/bot{self.config.bot_token}"
+
+    def instance_init(self):
+        """This instance init method is called automatically when an instance of this package is created. It registers the URL of the instance as the Telegram webhook for messages."""
+        webhook_url = self.context.invocable_url + "respond"
+        response = requests.get(
```

Comment:
It might even be worth it to go one level MORE error-tolerant and wrap this whole thing in a try/catch that just complains but otherwise continues.

That way we can confidently support folks who never enter a telegram key (my thinking: this is probably about to become the *de facto* base class for ALL AGENTS until we have something to replace it with, so planning for the case where someone doesn't want to involve telegram is probably good)

Reply:
I'm not sure I follow this one. If we're in the context of inheriting from `TelegramBot` (not `Agent` or anything more general) then I would argue that an instance that doesn't provide a valid telegram bot key is most definitely an error.